com.unity.ml-agents/Runtime/Actuators/ActuatorManager.cs

```diff
@@ -99,10 +101,40 @@ internal void ReadyActuatorsForExecution(IList<IActuator> actuators, int numCont
             var discreteActions = numDiscreteBranches == 0 ? ActionSegment<int>.Empty : new ActionSegment<int>(new int[numDiscreteBranches]);
 
             StoredActions = new ActionBuffers(continuousActions, discreteActions);
-            m_DiscreteActionMask = new ActuatorDiscreteActionMask(actuators, sumOfDiscreteBranches, numDiscreteBranches);
+            m_CombinedActionSpec = CombineActionSpecs(actuators);
+            m_DiscreteActionMask = new ActuatorDiscreteActionMask(actuators, sumOfDiscreteBranches, numDiscreteBranches, m_CombinedActionSpec.BranchSizes);
             m_ReadyForExecution = true;
         }
 
+        internal static ActionSpec CombineActionSpecs(IList<IActuator> actuators)
```

Comment:
I originally had it as a static method like in `ActionSpec` like `static ActionSpec Combine(IList<ActionSpec>)`, but not sure if there's a good way of going from the List of Actuators in ReadyActuatorsForExecution() to the list of ActionSpecs